internal/sessiontest/redis_test.go

```diff
@@ -0,0 +1,259 @@
+package sessiontest
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/alicebob/miniredis"
+	irma "github.com/privacybydesign/irmago"
+	"github.com/privacybydesign/irmago/internal/test"
+	"github.com/privacybydesign/irmago/server"
+	"github.com/privacybydesign/irmago/server/requestorserver"
+	"github.com/stretchr/testify/require"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"testing"
+)
+
+func startRedis(t *testing.T) *miniredis.Miniredis {
+	mr, err := miniredis.Run()
+	require.NoError(t, err)
+	return mr
+}
+
+func redisConfigDecorator(mr *miniredis.Miniredis, fn func() *requestorserver.Configuration) func() *requestorserver.Configuration {
```

Comment:
Why does this accept and return a function, instead of just a `*requestorserver.Configuration`?

Reply:
Using a function makes sure that a fresh configuration gets used. So after having run a Redis-specific test the decorated configuration will not overwrite the default configuration for the other tests. We could use a pointer instead of a function but then we would need to manually revert the changes after each test.

It does however not solve the problem that you cannot run the tests in parallel.

Do you have a suggestion for a better solution? Possibly one that will enable running tests in parallel.